src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -12,37 +12,61 @@
 namespace Symfony\Component\Serializer\Exception;
 
 /**
- * @deprecated since Symfony 6.3, use {@see MissingConstructorArgumentException} instead
- *
  * @author Maxime VEBER <maxime.veber@nekland.fr>
  */
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
-     * @var string[]
+     * @var string
+     */
+    private $class;
+
+    /**
+     * @var MissingConstructorArgumentException[]
      */
-    private $missingArguments;
+    private $missingConstructorArgumentExceptions = [];
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    /**
+     * @param MissingConstructorArgumentException[]     $missingArguments
+     */
+    public function __construct(array $missingArguments)
```

Comment:
Changing the constructor signature is a BC break, why not just add a new optional argument:

```php
public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [], string $class = null)
```